src/main/java/com/google/firebase/auth/FirebaseTokenVerifierImpl.java

```diff
@@ -278,6 +283,18 @@ private boolean containsLegacyUidField(IdToken.Payload payload) {
     return false;
   }
 
+  private void checkTenantId(final FirebaseToken firebaseToken) throws FirebaseAuthException {
+    String tokenTenantId = Strings.nullToEmpty(firebaseToken.getTenantId());
```

Comment:
Nit: nullToEmpty is not strictly needed here, since you call equals on `this.tenantId` which is never null. But I'll leave it to you decide.

Reply:
We actually do need it or else we'd say there was a mismatch if both were initially null. I confirmed this theory by making the change and watching `FirebaseTokenVerifierImplTest.testVerifyToken` fail.